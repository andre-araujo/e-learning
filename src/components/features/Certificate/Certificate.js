import React, { Component } from 'react';

import {
  Text, Button,
} from '../../elements';

class Certificate extends Component {
  onSubmit = (e) => {
    e.preventDefault();
    const {
      courseId,
      getCertification,
    } = this.props;

    getCertification(courseId);
  }

  render() {
    const {
      isAdmin,
    } = this.props;

    return isAdmin ? null : (
      <div>
        <Text.Subtitle tag="h1" margin="0 0 10px">
          Certificado digital:
        </Text.Subtitle>

        <hr />

        <form onSubmit={this.onSubmit}>
          <Button full>
            Requisitar certificado
          </Button>
        </form>
      </div>
    );
  }
}

export default Certificate;
