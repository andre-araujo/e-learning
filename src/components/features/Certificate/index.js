import { connect } from 'react-redux';

import Certificate from './Certificate';
import { actions } from '../../../lib';

const mapStateToProps = ({ courseDetail }) => ({
  courseDetail,
});

const mapDispatchToProps = dispatch => ({
  getCertification: data => dispatch(actions.getCertification(data)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Certificate);
