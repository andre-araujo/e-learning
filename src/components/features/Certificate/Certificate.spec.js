import React from 'react';
import { mount } from 'enzyme';

import Certificate from './Certificate';

describe('Certificate component', () => {
  it('should match snapshot', () => {
    expect(mount(<Certificate />)).toMatchSnapshot();
  });
});
