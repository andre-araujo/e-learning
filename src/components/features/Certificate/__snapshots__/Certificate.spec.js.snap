// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Certificate component should match snapshot 1`] = `
<div>
  <Component
    margin="0 0 10px"
    tag="h1"
  >
    Certificado digital:
  </Component>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <Button
      disabled={false}
      full={true}
      gap={0}
      minWidth={null}
      secondary={false}
    >
      Requisitar certificado
    </Button>
  </form>
</div>
`;
